tests/react-test-app/run-android.sh

```diff
@@ -5,35 +5,14 @@ set -e
 
 PATH="/opt/android-sdk-linux/platform-tools:$PATH"
 
-if [ -n "$REALM_BUILD_ANDROID" ]; then
-    echo "Realm is already installed" # by inoking test.sh previously for example
-
-else
-    # Inform the prepublish script to build Android modules.
-    REALM_BUILD_ANDROID=1 npm install realm realm-tests
-fi
-
-
-cp ../../src/object-store/tests/query.json node_modules/realm-tests/query-tests.json
-
 if adb shell pm list packages | grep -q io.realm.react.testapp; then
     echo "Uninstalling old apk"
     adb uninstall io.realm.react.testapp
 fi
 
-echo "Reversing port for physical device"
-adb reverse tcp:8081 tcp:8081
-
-echo "Reversing port for Realm Object Server"
-adb reverse tcp:9080 tcp:9080
-
-echo "Building Release APK"
-pushd android && ./gradlew assembleRelease
-
-echo "Installing APK"
-adb install app/build/outputs/apk/release/app-release.apk
-
-echo "Starting the Main Activity"
-adb shell am start -n io.realm.react.testapp/.MainActivity
+echo "Running test application"
+npx react-native run-android
 
-popd
+# granting permissions after react-native run-android since the package needs to be installed
+echo "Granting permission to write to sdcard"
```

Comment:
Might be worth mentioning the need for this? Is it to write the XML output of running the test?

Reply:
isn't it obvious. There is a comment and a echo output.

Reply:
It's not obvious why this is needed at all?

Reply:
this says literally "granting permission to write to sdcard". I think its obvious that the application needs to have this permission

Reply:
But why do we need permission to write to the sdcard, would be my question when I read this code 2 months from now.

Reply:
that's in the code. if code is refactored to require sdcard permission or not in the future this permission and this script may stay not causing any issues. Also this permission is present in the manifest file but we are not writing comments there as well. 
In my view there is little value in adding another comment next to these two lines.

the first comment says why we are doing that after app install and the `echo` writes what it does for logging. Its perfect imo. If we say "Giving permission to write to sdcard so we can write our test output file there" means we are setting code behavior there. Then if code is refactored later to also write something else we need to remember to change this comment as well (which is in another file and so on). In my view that's not needed and is of little value

Reply:
I added a comment there. I still think that adds little value.